packages/es-components/src/components/containers/banner/useBanner.js

```diff
@@ -0,0 +1,59 @@
+/* eslint react/prop-types: 0 */
+import React from 'react';
+import styled from 'styled-components';
+import { useTheme } from '../../util/useTheme';
+
+const ContentWrapper = styled.div`
+  align-self: center;
+  flex-grow: 1;
+  word-break: break-word;
+`;
+
+function BannerContent(props) {
+  const { children, ...rest } = props;
+  return (
+    <>
+      <ContentWrapper {...rest}>{children}</ContentWrapper>
+    </>
+  );
+}
+
+const Banner = styled.div`
+  align-items: center;
+  background-color: ${props => props.variant.bgColor};
+  border-radius: 2px;
+  color: ${props => props.variant.textColor};
+  display: flex;
+  margin-bottom: 25px;
+  padding: 15px;
+
+  ${props =>`
+    a {
+      color: ${props.variant.textColor};
+
+      &:hover {
+        text-decoration: none;
+      }
+    }
+  `}
+
+  button[aria-expanded] {
+    color: ${props => props.variant.textColor};
+  }
+`;
+
+export function useBanner() {
+  return function BaseBanner({ type, children, className, style, ...rest }) {
+    const theme = useTheme();
+    const color = theme.bannerStyles[type];
+    const bannerContentProps = {
+      color,
+      ...rest
+    };
+    return (
+      <Banner variant={color} className={className} style={style}>
+        <BannerContent {...bannerContentProps}>{children}</BannerContent>
+      </Banner>
+    );
+  };
+}
```

Comment:
There's quite a bit here that seems unnecessary. First, a tip: we should _never_ return a scoped function in a hook unless it's defined statically (outside the scope of our hook), or defined using `useCallback` so it doesn't get _redefined_ on every render. This can (and will) cause infinite render loops. (Also, `useBanner` is not actually a hook, so it shouldn't be called `useBanner`.)

A few other things:
1. We don't need an inner container for the Banner. Either the downstream team should provide a single container for their content, or they should rely on `BannerContainer`'s layout styles to handle it for them. `ContentWrapper` isn't really supplying anything helpful enough for the complexity it adds to the implementation.
2. We should use `React.forwardRef` any time we're wrapping a styled component so that downstream teams can use refs as desired.
3. `useBanner` is not something we need at all. We just need to export a component.
4. After these changes, we can entirely remove the current Banner.js file, and rename this to Banner.js. Having two separate files for this implementation adds confusion for the reader, and it will no longer be necessary.

```suggestion
import React from 'react';
import styled from 'styled-components';
import { useTheme } from '../../util/useTheme';

const BannerContainer = styled.div`
  align-items: center;
  background-color: ${props => props.variant.bgColor};
  border-radius: 2px;
  color: ${props => props.variant.textColor};
  display: flex;
  padding: ${({theme}) => theme.spacing.defaultMargin})

  ${props =>`
    a {
      color: ${props.variant.textColor};

      &:hover {
        text-decoration: none;
      }
    }
  `}

  button[aria-expanded] {
    color: ${props => props.variant.textColor};
  }
`;

const ForwardedBanner = React.forwardRef(function Banner({ type, ...props }, ref) {
    const theme = useTheme();
    const variant = theme.bannerStyles[type];
    const bannerProps = { variant, ...props };
    return (<BannerContainer ref={ref} {...bannerProps} />)l
}

ForwardedBanner.propTypes = {
  /** The type of notification to render */
  type: PropTypes.oneOf(['success', 'info', 'warning', 'danger', 'advisor'])
    .isRequired
};

export default ForwardedBanner;
```

Reply:
Is necessary to use the name ForwardedBanner instead of just Banner?

Reply:
It's not necessary. I would say you could call it `Banner` and the function _inside_ the `forwardRef` call could be called something like `InnerBanner`. If I remember correctly, the lint rules want you to have a name on the function inside that `forwardRef` call.